platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -14,92 +14,77 @@
 package org.codice.ddf.platform.filter.delegate;
 
 import java.io.IOException;
-import java.util.Collection;
 import java.util.Collections;
-import java.util.Iterator;
-import java.util.LinkedList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
 
 import javax.servlet.Filter;
 import javax.servlet.FilterChain;
 import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 
-import org.osgi.framework.Bundle;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.filter.SecurityFilter;
+import org.ops4j.pax.web.extender.whiteboard.ExtenderConstants;
+import org.ops4j.pax.web.extender.whiteboard.internal.util.ServicePropertiesUtils;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.Constants;
-import org.osgi.framework.FrameworkUtil;
-import org.osgi.framework.InvalidSyntaxException;
 import org.osgi.framework.ServiceReference;
+import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * {@link DelegateServletFilter} is meant to detect any ServletFilters
- * and call their {@link FilterChain}. If none are found it becomes a pass
- * through filter.
- *
+ * {@link DelegateServletFilter} is meant to detect any {@link SecurityFilter}s and doFilter through
+ * those before continuing on its {@link FilterChain}. If no {@link SecurityFilter}s are found, it
+ * becomes a pass-through filter.
  */
 public class DelegateServletFilter implements Filter {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DelegateServletFilter.class);
 
-    private FilterConfig filterConfig;
+    final TreeSet<ServiceReference<SecurityFilter>>
+            sortedAndInitializedSecurityFilterServiceReferences;
 
-    protected BundleContext getContext() {
-        Bundle cxfBundle = FrameworkUtil.getBundle(DelegateServletFilter.class);
-        if (cxfBundle != null) {
-            return cxfBundle.getBundleContext();
-        }
-        return null;
+    // used to only init SecurityFilters after the DelegateServletFilter has been initialized with a filterConfig
+    final Set<ServiceReference<SecurityFilter>> notYetInitializedSecurityFilterServiceReferences;
+
+    private Boolean hasBeenInitialized;
+
+    private final BundleContext bundleContext;
+
+    FilterConfig filterConfig;
+
+    public DelegateServletFilter(final BundleContext bundleContext) {
```

Comment:
You cannot pass the bundle context in from blueprint and hold onto it like this. These bundle contexts go stale, which is why we were using FrameworkUtil previously. You can use and discard a bundle context within a constructor, but you cannot save the reference.